### ISSUE TITLE
Comment regex fails with strings containing a # character

### ISSUE THREAD

**Issue by xardnaxelax** [DiffSK/configobj#157]
config.ini
```
[foo]
bar = ['lala', 'lele', 'here comes the # but this is not a comment']
```

Example:
`mat = self._nolistvalue.match(value)`
results in the following matching groups:
```
0: ['lala', 'lele', 'here comes the 
1:  but this is not a comment']
```

Replacing
`(?:[^'"\#].*?)|     # unquoted`
with
`(?:[^'"\#].*)|     # unquoted`
resolves the problem BUT leads to not correctly regexing real comments.

No idea how to easily fix that without rewriting the whole regex.

**Comment by robdennis** [comment]
I agree that ``re`` here seems brittle and I think someone should probably do something after 5.1.0 is shipped
